enterprise/server/workflow/service/service.go

```diff
@@ -268,6 +274,142 @@ func (ws *workflowService) GetWorkflows(ctx context.Context, req *wfpb.GetWorkfl
 	return rsp, nil
 }
 
+func (ws *workflowService) ExecuteWorkflow(ctx context.Context, req *wfpb.ExecuteWorkflowRequest) (*wfpb.ExecuteWorkflowResponse, error) {
+	// Validate req
+	if req.GetWorkflowId() == "" {
+		return nil, status.InvalidArgumentError("Missing workflow_id")
+	}
+	if req.GetCommitSha() == "" {
+		return nil, status.InvalidArgumentError("Missing commit_sha")
+	}
+	if req.GetBranch() == "" {
+		return nil, status.InvalidArgumentError("Missing branch")
+	}
+	if req.GetActionName() == "" {
+		return nil, status.InvalidArgumentError("Missing action_name")
+	}
+
+	// Authenticate
+	user, err := perms.AuthenticatedUser(ctx, ws.env)
+	if err != nil {
+		return nil, err
+	}
+
+	// Lookup workflow
+	wf := &tables.Workflow{}
+	db := ws.env.GetDBHandle()
+	err = db.Raw(
```

Comment:
can you use:

 err := env.GetDBHandle().Transaction(ctx, func(tx *db.DB) error {

instead please?

Reply:
Hm, why is a transaction needed if there's only a single query executed?